evalml/data_checks/id_columns_data_check.py

```diff
@@ -57,14 +57,14 @@ def validate(self, X, y=None):
         }
 
         X = _convert_to_woodwork_structure(X)
-        X = _convert_woodwork_types_wrapper(X.to_dataframe())
 
-        col_names = [col for col in X.columns.tolist()]
+        col_names = [col for col in X.columns]
         cols_named_id = [col for col in col_names if (str(col).lower() == "id")]  # columns whose name is "id"
         id_cols = {col: 0.95 for col in cols_named_id}
 
-        non_id_types = ['float16', 'float32', 'float64', 'bool']
-        X = X.select_dtypes(exclude=non_id_types)
+        X = X.select(include=['Integer', 'Categorical'])
```

Comment:
I think this makes sense!

Reply:
Agreed! Much easier to understand.

Does our documentation for this data check make it clear that this is the behavior?